### PR TITLE
test: Add test for ALTER SINK

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
Add test

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

**Comment by jkosh44** [comment]
I'm scratching my head here trying to figure out why this worked but the other test didn't. We should still look into why the other didn't but this also gives us some more test coverage which is probably good.

**Comment by benesch** [comment]
> I'm scratching my head here trying to figure out why this worked but the other test didn't. We should still look into why the other didn't but this also gives us some more test coverage which is probably good.

The old test had a silly bug. 😄 

```diff
diff --git a/test/testdrive/alter-sink.td b/test/testdrive/alter-sink.td
index d118c3c124..f8b424c950 100644
--- a/test/testdrive/alter-sink.td
+++ b/test/testdrive/alter-sink.td
@@ -125,8 +125,8 @@ contains:canceling statement due to user request
 # wait a few seconds more for good measure to avoid flaking.
 $ sleep-is-probably-flaky-i-have-justified-my-need-with-a-comment duration=4s
 
-> INSERT INTO post_alter VALUES ('hundred', 99);
+> INSERT INTO created_post_alter VALUES ('hundred', 99);
 
 # TODO(jkosh44) uncomment when we figure out why this doesn't work.
-# $ kafka-verify-data format=json sink=materialize.public.sink key=false
-# {"before": null, "after": {"created_post_name": "hundred", "created_post_value": 99}}
+$ kafka-verify-data format=json sink=materialize.public.sink key=false
+{"before": null, "after": {"created_post_name": "hundred", "created_post_value": 99}}
```

**Review by benesch — APPROVED** [review]
LGTM, but maybe better to use the more surgical patch I just commented? That tests altering the same sink multiple times, which is potentially valuable.

**Comment by jkosh44** [comment]
> The old test had a silly bug. 😄

LOL

**Review by benesch — APPROVED** [review]

File: test/testdrive/alter-sink.td [hunk 1]
```diff
@@ -125,8 +125,7 @@ contains:canceling statement due to user request
 # wait a few seconds more for good measure to avoid flaking.
 $ sleep-is-probably-flaky-i-have-justified-my-need-with-a-comment duration=4s
 
-> INSERT INTO post_alter VALUES ('hundred', 99);
+> INSERT INTO created_post_alter VALUES ('hundred', 99);
```
Comment (def-) [inline test/testdrive/alter-sink.td:128]: Sorry about that!

**Comment by benesch** [comment]
Not at all! Thank you for all the help testing this.

**Pull request merged**